### DISCUSSION TITLE
mastodon-sync-to-memos/

### DISCUSSION THREAD

**Discussion by giscus** [eallion/eallion.com#109] [Giscus]
# mastodon-sync-to-memos/

文章介绍了如何使用 Shell 脚本将 Mastodon 上的内容同步到 Memos。作者使用 Shell 脚本实现了这一功能，并提供了详细的配置和操作步骤。脚本会检查最新的 Mastodon 内容，并将其发布到 Memos 上，同时记录 Mastodon ID 和 Memos ID 的绑定关系。同时，作者还分享了自己在使用过程中遇到的问题和解决方案,并给出了具体示例和代码。

https://www.eallion.com/mastodon-sync-to-memos/

**Comment by jkjoy** [comment]
这个好~~~嘿嘿嘿

**Reply by eallion** [reply]
就是原来的脚本，完善了一下，发出来了。

**Comment by skyue** [comment]
Memos越来越难用了，练手项目很贴切。

**Reply by eallion** [reply]
确实有很多想不到的。理解不能。

**Reply by jkjoy** [reply]
我反而觉得是因为很多人提了很多要求导致memos变了样了。
或者说他本身就没有很清晰，合理的定位

**Reply by eallion** [reply]
有这个原因吧！

**Comment by xinkeji** [comment]
总结了下，最近研究大老日常，蜗牛大佬主页优化负优化了，打开速度非常慢，不知道是不是我网速原因反正就很慢 ，，第二个你目前用的图片封面手机版用下载不了，就很头疼，大佬可以修复下不，第三个标题自动生成og图的可以使用链接字体，不需要那么麻烦，亲测，嘿嘿

**Reply by eallion** [reply]
非常感谢您的反馈！
1. 我把首页的视频背景和 Polyfill 去掉了，其他的全是本地资源了。可能这个问题会有所缓解。
2. 我也是 fork 别人的在用，如果在手机上用，你可以用用原来的 https://github.com/youngle316/cover-paint ，稍后研究下手机版是什么问题。
3. OG Image 原来是用的 API，现在会有首页和其他页面调用 OG Image 就手动生成了。
再次感谢！

**Comment by koobai** [comment]
memos我就停留再 0.18.2 了，不去折腾它了

**Reply by eallion** [reply]
我现也是停留在这个版了，不过偶尔会手贱尝试升级试试。

**Reply by jkjoy** [reply]
如果单纯作为记录工具,memos 0.14.4是最好的.我认为此版本无论界面功能都做到了刚刚好.

0.18.2则是功能基本完善的版本.

**Comment by xinkeji** [comment]
这下快多了，不过还是有一小段空白，可以接受，之前有个30秒空白



---Original---
From: "Charles ***@***.***&gt;
Date: Tue, Mar 12, 2024 16:49 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] mastodon-sync-to-memos/ (Discussion #109)




 
有这个原因吧！

**Reply by eallion** [reply]
你说的空白是什么样的？

**Comment by xinkeji** [comment]
现在速度很快了截图不出来，不影响了



---Original---
From: "Charles ***@***.***&gt;
Date: Tue, Mar 12, 2024 22:01 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] mastodon-sync-to-memos/ (Discussion #109)




 
你说的空白是什么样的？

**Reply by eallion** [reply]
但是 30s 也太慢了，有机会帮忙反馈一下，我用工具测试也还行。

**Comment by xinkeji** [comment]
这个还有嘀咕也会很卡，你有缓存可能看不见

---Original---
From: "Charles ***@***.***&gt;
Date: Tue, Mar 12, 2024 22:02 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] mastodon-sync-to-memos/ (Discussion #109)




 
我现也是停留在这个版了，不过偶尔会手贱尝试升级试试。

**Reply by eallion** [reply]
嘀咕卡的话就正常，是从 Cloudflare 上动态加载的。

**Comment by xinkeji** [comment]
现在好像好很多了



---Original---
From: "Charles ***@***.***&gt;
Date: Tue, Mar 12, 2024 22:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] mastodon-sync-to-memos/ (Discussion #109)




 
但是 30s 也太慢了，有机会帮忙反馈一下，我用工具测试也还行。

**Comment by xinkeji** [comment]
我有审查过元素，大致瞟了一眼，有几个资源失效，我用的你源码，直接一堆资源失效，我直接换了dolt，原本打算用这个写说说的但是好像不太理想，就基本一直没有动，hugo果然是最折腾的框架，不过博客其实最快乐的就是折腾的过程



---Original---
From: "Charles ***@***.***&gt;
Date: Tue, Mar 12, 2024 22:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] mastodon-sync-to-memos/ (Discussion #109)




 
但是 30s 也太慢了，有机会帮忙反馈一下，我用工具测试也还行。

**Comment by lehhair** [comment]
对其他用户的回复你做了同步吗，我的misskey同步memos的脚本只对我自己发的内容和回复我自己的内容做同步，考虑到隐私问题，我没同步回复别人的内容了，而且引用内容也是个麻烦

**Reply by eallion** [reply]
```
# 跳过回复和转嘟
SKIP_MASTODON_REPLY=true
SKIP_MASTODON_REBLOG=true
```
默认是跳过的，Mastodon API 有这两个参数。
